server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundSummaryServiceTests.java

```diff
@@ -0,0 +1,251 @@
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.MockLog;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.Before;
+
+public class AllocationBalancingRoundSummaryServiceTests extends ESTestCase {
+    private static final Logger logger = LogManager.getLogger(AllocationBalancingRoundSummaryServiceTests.class);
+
+    final Settings enabledSummariesSettings = Settings.builder()
+        .put(AllocationBalancingRoundSummaryService.ENABLE_BALANCER_ROUND_SUMMARIES_SETTING.getKey(), true)
+        .build();
+    final Settings disabledDefaultEmptySettings = Settings.builder().build();
+    final Settings enabledButNegativeIntervalSettings = Settings.builder()
+        .put(AllocationBalancingRoundSummaryService.ENABLE_BALANCER_ROUND_SUMMARIES_SETTING.getKey(), true)
+        .put(AllocationBalancingRoundSummaryService.BALANCER_ROUND_SUMMARIES_LOG_INTERVAL_SETTING.getKey(), TimeValue.MINUS_ONE)
+        .build();
+
+    ClusterSettings enabledClusterSettings = new ClusterSettings(enabledSummariesSettings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+    ClusterSettings disabledDefaultEmptyClusterSettings = new ClusterSettings(
+        disabledDefaultEmptySettings,
+        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS
+    );
+    ClusterSettings enabledButNegativeIntervalClusterSettings = new ClusterSettings(
+        enabledButNegativeIntervalSettings,
+        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS
+    );
+
+    // Construction parameters for the service.
+
+    DeterministicTaskQueue deterministicTaskQueue;
+    ThreadPool testThreadPool;
+
+    @Before
+    public void setUpThreadPool() {
+        deterministicTaskQueue = new DeterministicTaskQueue();
+        testThreadPool = deterministicTaskQueue.getThreadPool();
+    }
+
+    /**
+     * Test that the service is disabled and no logging occurs when
+     * {@link AllocationBalancingRoundSummaryService#ENABLE_BALANCER_ROUND_SUMMARIES_SETTING} defaults to false.
+     */
+    public void testServiceDisabledByDefault() {
+        var service = new AllocationBalancingRoundSummaryService(testThreadPool, disabledDefaultEmptyClusterSettings);
+
+        try (var mockLog = MockLog.capture(AllocationBalancingRoundSummaryService.class)) {
+            /**
+             * Add a summary and check it is not logged.
+             */
+
+            service.addBalancerRoundSummary(new BalancingRoundSummary(50));
+            service.verifyNumberOfSummaries(0); // when summaries are disabled, summaries are not retained when added.
+            mockLog.addExpectation(
+                new MockLog.UnseenEventExpectation(
+                    "Running balancer summary logging",
+                    AllocationBalancingRoundSummaryService.class.getName(),
+                    Level.INFO,
+                    "Balancing round summaries:*"
+                )
+            );
+
+            deterministicTaskQueue.runAllRunnableTasks();
```

Comment:
Do we need to `advanceTime` here also? so if there were a scheduled task in the queue it would be executed?

Reply:
Thanks for catching this, I should have looked more closely.

advanceTime [throws](https://github.com/elastic/elasticsearch/blob/3fafb5f1610cb6702a9643d2df816c1db26e61af/test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java#L209) because no tasks were ever scheduled. It looks like I can check for deferred (delayed) tasks though 🤔  So I've added that check before running advance: I think this should catch if something was unexpectedly scheduled.

runTasksUpToTimeInOrder was another option: set a future time 2x the logging interval, say. But I think checking deferred is cleaner.